server/src-lib/Hasura/RQL/DDL/RemoteRelationship/Validate.hs

```diff
@@ -409,7 +412,16 @@ isTypeCoercible actualType expectedType =
   let (actualBaseType, actualNestingLevel) = getBaseTyWithNestedLevelsCount actualType
       (expectedBaseType, expectedNestingLevel) = getBaseTyWithNestedLevelsCount expectedType
   in
-  if | actualBaseType /= expectedBaseType -> raiseValidationError
+  if | expectedBaseType == $$(G.litName "ID") ->
+         bool (throwError $ IDTypeJoin actualBaseType)
+              (pure ())
+              -- Check under `Input Coercion` https://spec.graphql.org/June2018/#sec-ID
+              -- We can also include the `ID` type in the below list but it will be
+              -- extraneous because at the time of writing this, we don't generate
+              -- the `ID` type in the DB schema
+              (G.unName actualBaseType `elem`
```

Comment:
We are being arbitrarily lax in our type system. So why not all scalars (in any case, we need `uuid`)? Should we make this a configuration element "coerce into ID type"?

Reply:
We should think what happens when hasura does support `ID` type. I guess the schema customisation roadmap will allow something like "make this column ID type" (and it will validate that the column has uniqueness constraint)

Reply:
I guess we can extend the coercible types here to custom scalar types, `Int` and `String` types? I'm consciously not supporting the `Boolean` and `Float` types because the spec says that when these values are received as input values, the remote schema should throw an error.

> When expected as an input type, any string (such as "4") or integer (such as 4) input value should be coerced to ID as appropriate for the ID formats a given GraphQL server expects. Any other input value, including float input values (such as 4.0), must raise a query error indicating an incorrect type.

Reply:
> Should we make this a configuration element "coerce into ID type"
Yes, I agree. That will be better, but will that only be for remote relationships or will the type of that column be changed in the schema also?

Reply:
I guess coercion is expected as per the spec. Lets just add `uuid` type to this list?